addons/binding/org.openhab.binding.gpstracker/src/main/java/org/openhab/binding/gpstracker/internal/discovery/TrackerDiscoveryService.java

```diff
@@ -0,0 +1,125 @@
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.gpstracker.internal.GPSTrackerBindingConstants;
+import org.openhab.binding.gpstracker.internal.config.ConfigHelper;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The {@link TrackerDiscoveryService} class provides discovery service for the binding to discover trackers. Discovery
+ * process is initiated by the tracker by sending a GPS log record. Based on the tracker id received in thin record an
+ * entry is created in the Inbox for the thing representing the tracker.
+ *
+ * @author Gabor Bicskei - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = { DiscoveryService.class, TrackerDiscoveryService.class }, immediate = true, configurationPid = "discovery.gpstracker")
+public class TrackerDiscoveryService extends AbstractDiscoveryService {
+    /**
+     * Discovery timeout
+     */
+    private static final int TIMEOUT = 1;
+
+    /**
+     * Registry of tracker to discover next time
+     */
+    private Set<String> trackersToDiscover = new HashSet<>();
+
+    /**
+     * Constructor.
+     *
+     * @throws IllegalArgumentException thrown by the super constructor
+     */
+    public TrackerDiscoveryService() throws IllegalArgumentException {
+        super(GPSTrackerBindingConstants.SUPPORTED_THING_TYPES_UIDS, TIMEOUT, true);
+    }
+
+    /**
+     * Called when the source tracker is not registered as a thing. These undiscovered trackers will be registered by
+     * the discovery service.
+     *
+     * @param trackerId Tracker id.
+     */
+    public void addTracker(String trackerId) {
+        trackersToDiscover.add(trackerId);
+        if (isBackgroundDiscoveryEnabled()) {
+            createDiscoveryResult(trackerId);
+        }
+    }
+
+    /**
+     * Unregister the tracker after the thing handles is created.
+     *
+     * @param trackerId Tracker id to unregister
+     */
+    public void removeTracker(String trackerId) {
+        trackersToDiscover.remove(trackerId);
+    }
+
+    @Override
+    protected void startScan() {
+        trackersToDiscover.forEach(this::createDiscoveryResult);
+    }
+
+    /**
+     * Create discovery result form the tracker id.
+     *
+     * @param trackerId Tracker id.
+     */
+    private void createDiscoveryResult(String trackerId) {
+        ThingUID id = new ThingUID(GPSTrackerBindingConstants.THING_TYPE_TRACKER, trackerId);
+        DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(id)
+                .withProperty(ConfigHelper.CONFIG_TRACKER_ID, trackerId)
+                .withThingType(GPSTrackerBindingConstants.THING_TYPE_TRACKER)
+                .withLabel("GPS tracker")
```

Comment:
Hm, that wasn't the change I meant. All I requested was a "T" instead of a "t", but not the removal of the tracker name...

Reply:
Oh. I misunderstood... Changing it back.